tools/psa/release.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/python
+# Copyright (c) 2017-2018 ARM Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+import sys
+import shutil
+
+ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), os.pardir, os.pardir))
+sys.path.insert(0, ROOT)
+from tools.targets import Target, TARGET_MAP, TARGET_NAMES
+
+PYTHON_EXEC = 'python' if os.path.basename(sys.executable).startswith('mbed') else sys.executable
+MAKE_PY_LOCATTION = os.path.join(ROOT, 'tools', 'make.py')
+TEST_PY_LOCATTION = os.path.join(ROOT, 'tools', 'test.py')
+MBED_PSA_TESTS = '*psa-spm*,*psa-crypto_access_control'
+TFM_MBED_APP = os.path.join(ROOT, 'tools', 'psa', 'tfm', 'mbed_app.json')
+TFM_TESTS = {
+    '*psa-spm_smoke': ['USE_PSA_TEST_PARTITIONS', 'USE_SMOKE_TESTS_PART1'],
+    '*psa-spm_client': ['USE_PSA_TEST_PARTITIONS', 'USE_CLIENT_TESTS_PART1'],
+    '*psa-spm_server': ['USE_PSA_TEST_PARTITIONS', 'USE_SERVER_TESTS_PART1', 'USE_SERVER_TESTS_PART2'],
+    '*psa-crypto_access_control': ['USE_PSA_TEST_PARTITIONS', 'USE_CRYPTO_ACL_TEST']
+}
+
+
+def get_mbed_official_psa_release():
+    def psa_backend(target_name):
```

Comment:
In general, I prefer to have functions outside of other functions, especially in the case here: you're not capturing anything from the calling function. If this function is private, prefix it with an `_`